### ISSUE TITLE
[ls] LS cannot receive DidChangeWatchedFilesNotification during its initialization

### ISSUE THREAD

**Issue by kittaakos** [eclipse/xtext-core#801]
I am not sure whether this is a bug or not, but in some rare situations, it can break the incremental builder and requires client adjustments for the Xtext LS.

**Iff** a language generates model files during the build, the client receives the new file events and tries to send a `DidChangeWatchedFilesNotification` to the server. But it cannot, it is still initializing, and the build phase is in progress. This will break the incremental build chain, and after the LS initialization, there is a half-baked workspace state.

Related thread: https://github.com/theia-ide/theia/issues/2420

**Comment by cdietrich** [comment]
do you generate files of one dsl files from another dsl file or is the files unrelated?

**Comment by kittaakos** [comment]
> in eclipse this generation of other dsl files wont work too.

It works. The Eclipse builder immediately kicks in with the new files. At least, that was my experience so far.

> postpone outgoing notifications until init is done
> queue incoming notifications until init is done

That is my local workaround so far.

> did you talk to miro or sven on that issue?

Not yet. I am sure we get back to these issues soon and I will update this thread.

**Comment by cdietrich** [comment]
> It works. The Eclipse builder immediately kicks in with the new files. At least, that was my experience so far.

then you have a second builder that touches the first files. a eclipse builder never sees the files it created itself.

**Comment by kittaakos** [comment]
> then you have a second builder that touches the first files

Exactly. This second thingy is missing in case of the LS because it cannot receive the notifications about the watched files changes.
